### PR TITLE
Add clang-8 builds for linux-next arm64-defconfig and x86_64-defconfig

### PR DESCRIPTION
**Pull request by mattface**
A reasonably safe hack to enable only arm64/x86 defconfig/allmodconfig clang builds for a tree called next-clang, which is a clone of linux-next.
All other trees (including original next) build all the same configs as before.

### CODE REVIEW

**Comment by gctucker** [comment]
Please make another work-around on top of #75 so that we can revert this patch and apply the new work-around when merging #75.

File: jenkins/build-trigger.jpl [hunk 1]
```diff
@@ -266,31 +264,43 @@ node("defconfig-creator") {
...
+    def arch_config_cc_ccv = []
 
     stage("Configs") {
         for (String arch: archs) {
             def configs = ["allnoconfig"]
+            def cc = 'gcc'
+            def cc_version = '7'
+
+            if (params.TREE_NAME == "next-clang") {
+                if (arch == "x86_64" || arch == "arm64") {
```
Comment (khilman) [inline jenkins/build-trigger.jpl]: Only arm64 is working with clang-8 currently.  Drop x86_64.

File: jenkins/kernel-arch-complete.sh [hunk 2]
```diff
@@ -197,6 +197,10 @@ if [[ BUILDS_FINISHED -eq 4 ]]; then
...
+    elif [ "$TREE_NAME" == "next-clang" ]; then
+        echo "Sending results to Linux Next (clang only)"
+        curl -X POST -H "Authorization: $EMAIL_AUTH_TOKEN" -H "Content-Type: application/json" -d '{"job": "'$TREE_NAME'", "kernel": "'$GIT_DESCRIBE'", "git_branch": "'$BRANCH'", "build_report": 1, "send_to": ["broonie@kernel.org", "arnd.bergmann@linaro.org", "ndesaulniers@google.com", "kernel-build-reports@lists.linaro.org"], "format": ["txt", "html"], "delay": 10}' ${API}/send
+        curl -X POST -H "Authorization: $EMAIL_AUTH_TOKEN" -H "Content-Type: application/json" -d '{"job": "'$TREE_NAME'", "kernel": "'$GIT_DESCRIBE'", "git_branch": "'$BRANCH'", "boot_report": 1, "send_to": ["broonie@kernel.org", "arnd.bergmann@linaro.org", "ndesaulniers@google.com", "kernel-build-reports@lists.linaro.org"], "format": ["txt", "html"], "delay": 12600}' ${API}/send
```
Comment (khilman) [inline jenkins/kernel-arch-complete.sh]: Why the HTML mails.  I thought we were removing those?

**Comment by mattface** [comment]
Dropping this PR as the hack will reside on staging until the YAML setup is ready for production.

**Pull request closed without merging**